edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent.Test/WorkerTests.cs

```diff
@@ -0,0 +1,348 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent.Test
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Reflection;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Azure.Devices.Edge.Util.Test.Common;
+    using Microsoft.Extensions.Logging.Abstractions;
+    using Moq;
+    using Xunit;
+
+    public class WorkerTests : IDisposable
+    {
+        TempDirectory tempDirectory = new TempDirectory();
+
+        [Fact]
+        public async Task TestScraping()
+        {
+            /* test data */
+            (string name, double value)[] modules = Enumerable.Range(1, 10).Select(i => ($"module_{i}", 1.0)).ToArray();
+
+            /* Setup mocks */
+            var scraper = new Mock<IMetricsScraper>();
+            scraper.Setup(s => s.ScrapeEndpointAsync(CancellationToken.None)).ReturnsAsync(() => this.PrometheousMetrics(modules));
+
+            var storage = new Mock<IMetricsStorage>();
+            string storedValue = string.Empty;
+            storage.Setup(s => s.WriteData(It.IsAny<string>())).Callback((Action<string>)(data => storedValue = data));
+
+            var uploader = new Mock<IMetricsUpload>();
+
+            MetricsWorker worker = new MetricsWorker(scraper.Object, storage.Object, uploader.Object);
+            MethodInfo methodInfo = typeof(MetricsWorker).GetMethod("Scrape", BindingFlags.NonPublic | BindingFlags.Instance);
```

Comment:
you don't need to use reflection, just make Scrape method as internal and add InternalToVisible parameter for DiagnosticComponentTest project in DiagnosticComponent project.  Therefore internal method in DiagnosticsComponent project can be accessed by DiagnosticsComponentTest project.